semgrep-core/src/core/Metavariable.ml

```diff
@@ -75,7 +75,10 @@ type mvalue =
    * where strings are often used to represent entities (e.g., function
    * names).
    *)
-  | Text of string AST_generic.wrap
+  | Text of
```

Comment:
Maybe a comment saying why we keep the original token with enclosing quotes.
I guess the reasoning is that when people write "$X", they don't care whether in the target
code they used 'foo' or "foo", and if we use an autofix with "$X" we want to reput the
exact same string ('foo' or "foo"). The alternative would require a way to match the specific
quote used for the string but this seems heavy.

Not sure though how it will solve autofix like stuff("bar $X").

Reply:
For something like `stuff("bar $X")` we'll have to synthesize a new string, escape the text, and print it out without lifting any original text. So, this doesn't help, but both of these ranges will be wrong and irrelevant since it will be a new synthetic AST node. I will add a comment though.